### PR TITLE
mongodb: version bump

### PR DESCRIPTION
**Pull request by D3vil0p3r**
Latest version that fixes the issue explained in https://github.com/BlackArch/blackarch/issues/3605

### CODE REVIEW

**Comment by D3vil0p3r** [comment]
Hey @noraj can you please review this mongoDB version bump? It should solve the issue related to old boost libraries.

**Review by noraj — CHANGES_REQUESTED** [review]

File: packages/mongodb/PKGBUILD [hunk 1]
```diff
@@ -17,6 +17,7 @@ _pkgname=mongodb
 pkgver=6.0.2
 pkgrel=1
 pkgdesc="A high-performance, open source, schema-free document-oriented database"
+groups=('blackarch' 'blackarch-misc')
```
[the remark this answers is not in the document]
Comment (D3vil0p3r) [reply packages/mongodb/PKGBUILD]: > We don't put groups to package that are just dependencies.

Ok. I changed the PKGBUILD by deleting `groups` row.

**Comment by D3vil0p3r** [comment]
One important note: the package must be compiled by 1 core. So it is needed to edit `/etc/makepkg.conf` and change "MAKEFLAGS" variable to `MAKEFLAGS="-j1"`.

**Comment by D3vil0p3r** [comment]
@noraj can you give a look to this, if this PR could be ready to be merged?

**Comment by noraj** [comment]
> One important note: the package must be compiled by 1 core. So it is needed to edit `/etc/makepkg.conf` and change "MAKEFLAGS" variable to `MAKEFLAGS="-j1"`. I needed to set also 8 GB of RAM on my VM for preventing compilation error like `fatal error: Killed signal terminated program cc1plus`.

I don't know how our build pipeline can handle that. Any idea @noptrix?

> @noraj can you give a look to this, if this PR could be ready to be merged?

I'm not confident reviewing that kind of package. Would be nice if someone more familiar with patching could review it.

File: packages/mongodb/PKGBUILD [hunk 2]
```diff
@@ -1,67 +1,91 @@
...
+source=(https://fastdl.mongodb.org/src/mongodb-src-r$pkgver.tar.gz
+        mongodb.sysusers
+        mongodb.tmpfiles
+        mongodb-4.4.1-gcc11.patch
+        mongodb-5.0.2-no-compass.patch
+        mongodb-5.0.2-skip-reqs-check.patch
+        mongodb-6.0.0-boost-1.79.patch
+        mongodb-6.0.2-sconstruct-changes.patch
+        mongodb-6.0.0-gcc12.patch)
+sha256sums=('39fa695a3a79008c818cb92df1fa716510d66069f7cec20a8fb3051286ac34bf'
```
Comment (noraj) [inline packages/mongodb/PKGBUILD:37]: all BA PKGBUILD use `sha512sums`

**Review by noraj — CHANGES_REQUESTED** [review]

File: packages/mongodb/PKGBUILD (hunk 2 above)
Comment (D3vil0p3r) [reply packages/mongodb/PKGBUILD]: Changed to sha512sums and version bumped to the latest one 6.0.4.

**Comment by D3vil0p3r** [comment]
> I'm not confident reviewing that kind of package. Would be nice if someone more familiar with patching could review it.

I updated the PKGBUILD for the latest MongoDB version and I tested it. It works correctly. The PKGBUILD code comes from MongoDB AUR package https://aur.archlinux.org/packages/mongodb

**Comment by noptrix** [comment]
@D3vil0p3r thank you

**Pull request merged**